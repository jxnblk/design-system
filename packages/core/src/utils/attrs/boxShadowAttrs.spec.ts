import { boxShadowAttrs, boxShadowSizeValues } from '.'
import { createTheme } from '..'

const theme = createTheme()

describe('boxShadowAttrs', () => {
  it.each(boxShadowSizeValues)('works with and boxShadowSize=%s', (boxShadowSize) => {
    const attrs = boxShadowAttrs({ boxShadowSize, theme })

    expect(attrs).toMatchSnapshot()
  })

  it('should handle an array prop', () => {
    const attrs = boxShadowAttrs({ boxShadowSize: [...boxShadowSizeValues, '9999px'], theme })

    expect(attrs).toMatchSnapshot()
  })
})
