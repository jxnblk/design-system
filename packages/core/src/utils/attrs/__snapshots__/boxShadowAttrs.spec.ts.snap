// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`boxShadowAttrs should handle an array prop 1`] = `
Object {
  "boxShadow": Array [
    "0 0 1px 0 rgba(0,0,0,0.24),0 2px 1px -1px rgba(0,0,0,0.16),0 2px 4px 0 rgba(0,0,0,0.12)",
    "0 0 2px 0 rgba(0,0,0,0.2),0 4px 2px -2px rgba(0,0,0,0.12),0 4px 8px -1px rgba(0,0,0,0.16)",
    "0 1px 4px 0 rgba(0,0,0,0.2),0 6px 4px -4px rgba(0,0,0,0.12),0 8px 16px -1px rgba(0,0,0,0.16)",
    "0 2px 8px 0 rgba(0,0,0,0.16),0 10px 8px -5px rgba(0,0,0,0.16),0 12px 32px -2px rgba(0,0,0,0.16)",
    "0 4px 12px 0 rgba(0,0,0,0.16),0 12px 12px -4px rgba(0,0,0,0.16),0 24px 64px -2px rgba(0,0,0,0.16)",
    "0 8px 32px 0 rgba(0,0,0,0.24),0 8px 16px 0 rgba(0,0,0,0.2),0 24px 64px 0 rgba(0,0,0,0.2)",
    "0 24px 72px 0 rgba(0,0,0,0.48),0 8px 16px 0 rgba(0,0,0,0.12),0 24px 64px 0 rgba(0,0,0,0.2)",
    undefined,
  ],
}
`;

exports[`boxShadowAttrs works with and boxShadowSize=2xl 1`] = `
Object {
  "boxShadow": "0 4px 12px 0 rgba(0,0,0,0.16),0 12px 12px -4px rgba(0,0,0,0.16),0 24px 64px -2px rgba(0,0,0,0.16)",
}
`;

exports[`boxShadowAttrs works with and boxShadowSize=lg 1`] = `
Object {
  "boxShadow": "0 1px 4px 0 rgba(0,0,0,0.2),0 6px 4px -4px rgba(0,0,0,0.12),0 8px 16px -1px rgba(0,0,0,0.16)",
}
`;

exports[`boxShadowAttrs works with and boxShadowSize=md 1`] = `
Object {
  "boxShadow": "0 0 2px 0 rgba(0,0,0,0.2),0 4px 2px -2px rgba(0,0,0,0.12),0 4px 8px -1px rgba(0,0,0,0.16)",
}
`;

exports[`boxShadowAttrs works with and boxShadowSize=overlay-lg 1`] = `
Object {
  "boxShadow": "0 8px 32px 0 rgba(0,0,0,0.24),0 8px 16px 0 rgba(0,0,0,0.2),0 24px 64px 0 rgba(0,0,0,0.2)",
}
`;

exports[`boxShadowAttrs works with and boxShadowSize=overlay-xl 1`] = `
Object {
  "boxShadow": "0 24px 72px 0 rgba(0,0,0,0.48),0 8px 16px 0 rgba(0,0,0,0.12),0 24px 64px 0 rgba(0,0,0,0.2)",
}
`;

exports[`boxShadowAttrs works with and boxShadowSize=sm 1`] = `
Object {
  "boxShadow": "0 0 1px 0 rgba(0,0,0,0.24),0 2px 1px -1px rgba(0,0,0,0.16),0 2px 4px 0 rgba(0,0,0,0.12)",
}
`;

exports[`boxShadowAttrs works with and boxShadowSize=xl 1`] = `
Object {
  "boxShadow": "0 2px 8px 0 rgba(0,0,0,0.16),0 10px 8px -5px rgba(0,0,0,0.16),0 12px 32px -2px rgba(0,0,0,0.16)",
}
`;
